### PR TITLE
BAU: Update govuk-rubocop to 3.6.0

### PR DESCRIPTION
**Pull request by alan-gds**
I've just noticed that we can now get rubocop-govuk to a higher version than we were able to back when I looked at linting over firebreak; this is probably a result of the linting fixes @koetsier committed recently.  This PR bumps it up to 3.6.0, requiring only one small fix in a .erb file along the way.  3.7.0 is blocked by an issue in another .erb, which I'm not sure how best to fix.  Let's get this update in, and then we can try and sort that out.

### CODE REVIEW

**Comment by alan-gds** [comment]
Damn, the build is failing, probably because of my change to the .erb file.

**Comment by alan-gds** [comment]
The .erb issue is now fixed and the PR is ready for review.

File: Gemfile [hunk 1]
```diff
@@ -37,7 +37,7 @@ group :test do
   gem "rack_session_access"
   gem "rails-controller-testing"
   gem "rspec-rails", "~> 4"
-  gem "rubocop-govuk"
+  gem "rubocop-govuk", "=3.6.0"
```
Comment (freesteph) [inline Gemfile:40]: can  we use `~> 3.6` here?

File: app/views/layouts/_flash_notices.html.erb [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
   <% if name == 'notice' %>
     <div class='flash-holder'>
-      <%= content_tag :p, msg, class: "govuk-heading-m flash-message-notice" %>
+      <%= tag.p(msg, class: "govuk-heading-m flash-message-notice") %>
```
Comment (robinmitra) [inline app/views/layouts/_flash_notices.html.erb:7]: @alan-gds Can you provide more context please as to why `tag.p` is preferred here?

File: Gemfile (hunk 1 above)
Comment (alan-gds) [reply Gemfile:40]: Eek, this was a mistake.  Well spotted.

File: app/views/layouts/_flash_notices.html.erb (hunk 2 above)
[the remark this answers is not in the document]
Comment (alan-gds) [reply app/views/layouts/_flash_notices.html.erb:7]: Against 3.6.0?  Is the linter definitely getting run against the .erb files on your machine?

Comment (alan-gds) [reply app/views/layouts/_flash_notices.html.erb:7]: Apparently content_tag is legacy syntax.
https://github.com/rubocop-hq/rubocop-rails/pull/242

[the remark this answers is not in the document]
Comment (freesteph) [reply app/views/layouts/_flash_notices.html.erb:7]: Yep, no offence here.

[the remark this answers is not in the document]
Comment (freesteph) [reply app/views/layouts/_flash_notices.html.erb:7]: My apologies, it is an offence, I ran `rubocop` instead of `erb-lint`.

**Comment by alan-gds** [comment]
I've fixed the problem Steph spotted, where rubocop-govuk was being pinned to a specific version.  It shouldn't be pinned at all; I committed that in error.

Hopefully this is all good to go now!

File: Gemfile (hunk 1 above)
Comment (freesteph) [reply Gemfile]: Do we just enforce in the lock file then?

Comment (freesteph) [reply Gemfile]: just saw the comment, all good :+1:

**Review by freesteph — APPROVED** [review]

**Pull request merged**